configuration.go

```diff
@@ -1,166 +1,169 @@
-package p2p
-
-import (
-	"time"
-)
-
-const (
-	// Broadcast sends a parcel to multiple peers (randomly selected based on fanout and special peers)
-	Broadcast = "<BROADCAST>"
-	// FullBroadcast sends a parcel to all peers
-	FullBroadcast = "<FULLBORADCAST>"
-	// RandomPeer sends a parcel to one randomly selected peer
-	RandomPeer = "<RANDOMPEER>"
-)
-
-// Configuration defines the behavior of the gossip network protocol
-type Configuration struct {

```

Comment:
Looks like we are hardwired to TCP.  Can we support UDP?  Would be great if we could support simulation nodes.  The current code puts simulation above quite a bit of logic in the gossip network that we *could* test if we made the protocol switchable.

Reply:
I'm not sure why you need UDP for simulation nodes. Everything in the factomd network is currently TCP based, so the only way for this to be a compatible drop-in replacement is also TCP. I've been simulating networks by running each simulated node on a different loopback ip (127.xxx.xxx.xxx), giving me over 16million potential nodes.

Looking into UDP support is something I've had on the long-term roadmap but it's not something I planned for the initial release. It would be interesting to see how well running a dual network (TCP and UDP) would work though.